stream.go

```diff
@@ -155,6 +160,11 @@ func (cs *clientStream) Trailer() metadata.MD {
 }
 
 func (cs *clientStream) SendMsg(m interface{}) (err error) {
+	cs.mu.Lock()
```

Comment:
Okay, I care about avoiding hitting the global bool each time more than saving a few bytes, so I'll add a bool to clientStream. Stand by...